frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/workspace_overview/overview_block_component.dart

```diff
@@ -0,0 +1,335 @@
+import 'package:appflowy/workspace/application/view/view_service.dart';
+import 'package:appflowy_backend/log.dart';
+import 'package:appflowy_backend/protobuf/flowy-error/errors.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/icon.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/view.pb.dart';
+import 'package:appflowy_editor/appflowy_editor.dart' hide Log;
+import 'package:collection/collection.dart';
+import 'package:dartz/dartz.dart' hide State;
+import 'package:easy_localization/easy_localization.dart' hide TextDirection;
+import 'package:flowy_infra_ui/style_widget/hover.dart';
+import 'package:flowy_infra_ui/widget/spacing.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class OverviewBlockKeys {
+  const OverviewBlockKeys._();
+
+  static const String type = 'overview';
+  static const String backgroundColor = blockComponentBackgroundColor;
+}
+
+SelectionMenuItem overviewItem = SelectionMenuItem.node(
+  name: LocaleKeys.document_selectionMenu_overview.tr(),
+  iconData: Icons.list_alt,
+  keywords: ['overview', 'workspace overview'],
+  nodeBuilder: (editorState, context) => overviewBlockNode(),
+  replace: (editorState, node) => node.delta?.isEmpty ?? false,
+);
+
+Node overviewBlockNode() {
+  return Node(
+    type: OverviewBlockKeys.type,
+  );
+}
+
+class OverviewBlockComponentBuilder extends BlockComponentBuilder {
+  OverviewBlockComponentBuilder({
+    super.configuration,
+  });
+
+  @override
+  BlockComponentWidget build(BlockComponentContext blockComponentContext) {
+    final node = blockComponentContext.node;
+    return OverviewBlockWidget(
+      key: node.key,
+      node: node,
+      configuration: configuration,
+      showActions: showActions(node),
+      actionBuilder: (context, state) => actionBuilder(
+        blockComponentContext,
+        state,
+      ),
+    );
+  }
+
+  @override
+  bool validate(Node node) => node.children.isEmpty;
+}
+
+class OverviewBlockWidget extends BlockComponentStatefulWidget {
+  const OverviewBlockWidget({
+    super.key,
+    required super.node,
+    super.showActions,
+    super.actionBuilder,
+    super.configuration = const BlockComponentConfiguration(),
+  });
+
+  @override
+  State<BlockComponentStatefulWidget> createState() =>
+      _OverviewBlockWidgetState();
+}
+
+class _OverviewBlockWidgetState extends State<OverviewBlockWidget>
+    with
+        BlockComponentConfigurable,
+        BlockComponentTextDirectionMixin,
+        BlockComponentBackgroundColorMixin {
+  @override
+  BlockComponentConfiguration get configuration => widget.configuration;
+
+  @override
+  EditorState get editorState => context.read<EditorState>();
+
+  @override
+  Node get node => widget.node;
+
+  final double leftIndentIncrementValue = 20.0;
+
+  @override
+  Widget build(BuildContext context) {
+    final viewId = editorState.document.id!;
```

Comment:
To avoid introducing "id" to a Document, you can do this instead:
```suggestion
    final viewId = context.read<DocumentBloc>().state.view.id;
```